### PR TITLE
Fix hessian criteria statistics collector

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Fix bug with passing subset_size

### Tests

Add a test for backend methods

### CODE REVIEW

**Comment by kshpv** [comment]
test_examples 606 - fixed num of samples for tiny_llama sample

File: tests/cross_fw/test_templates/test_weights_compression_backends.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def get_hawq_with_backend(self, subset_size: int):
+        """
+        Returns an instance of the algorithm with the specified subset size.
+
+        :param subset_size: The size of the subset to be used by the algorithm.
+        :return: An instance of the algorithm.
+        """
+
+    @pytest.mark.parametrize("subset_size", (10, 1, None))
+    def test_hawq_statistic_collector(self, subset_size: int):
```
Comment (ljaljushkin) [inline tests/cross_fw/test_templates/test_weights_compression_backends.py:32]: It's good to have tests for hawq stat collector. What about adding tests for others?

There are already tests for calling statistics and passing subset_size. https://github.com/openvinotoolkit/nncf/blob/develop/tests/openvino/native/quantization/test_weights_compression.py#L892-L918
you could just extend them with list of sensitivity criterion.

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: tests/openvino/native/quantization/test_weights_compression_backends.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from nncf.quantization.algorithms.weight_compression.mixed_precision import HAWQCriterion
+from nncf.quantization.algorithms.weight_compression.openvino_backend import OVMixedPrecisionAlgoBackend
+from tests.cross_fw.test_templates.test_weights_compression_backends import TemplateTestMixedPrecisionAlgoBackend
+from tests.openvino.native.models import IdentityMatmul
+
+
+class TestOVMixedPrecisionAlgoBackend(TemplateTestMixedPrecisionAlgoBackend):
+    def get_hawq_with_backend(self, subset_size):
+        hawq = HAWQCriterion(None, None, subset_size=subset_size)
+        hawq._backend_entity = OVMixedPrecisionAlgoBackend(IdentityMatmul().ov_model)
```
[the remark this answers is not in the document]
Comment (ljaljushkin) [reply tests/openvino/native/quantization/test_weights_compression_backends.py:20]: Suppose it's coming from base class `OVWeightCompressionAlgoBackend`
@nikita-savelyevv could you comment?

File: nncf/quantization/algorithms/weight_compression/mixed_precision.py [hunk 3]
```diff
@@ -360,7 +360,7 @@ def _calc_weight_sensitivity(
         return fns.linalg.norm(decompressed_weight - weight, ord="fro").item()
 
     def _get_statistic_collector(self):
-        return self._backend_entity.hawq_statistic_collector()
+        return self._backend_entity.hawq_statistic_collector(self._subset_size)
```
Comment (nikita-savelyevv) [inline nncf/quantization/algorithms/weight_compression/mixed_precision.py:363]: While we are here. Could you please add back a `self` argument to the definition of the base `_get_statistic_collector` method? I've missed this during the last review.
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/weight_compression/mixed_precision.py#L284

**Review by nikita-savelyevv — COMMENTED** [review]
👍

File: tests/openvino/native/quantization/test_weights_compression_backends.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kshpv) [reply tests/openvino/native/quantization/test_weights_compression_backends.py:20]: I am just confused about the general idea of inheritance from OVWeightCompressionAlgoBackend, while most of the part of the inherited class is not used
we can discuss it offline

File: nncf/quantization/algorithms/weight_compression/mixed_precision.py (hunk 3 above)
Comment (kshpv) [reply nncf/quantization/algorithms/weight_compression/mixed_precision.py:363]: sure

File: tests/cross_fw/test_templates/test_weights_compression_backends.py (hunk 1 above)
Comment (kshpv) [reply tests/cross_fw/test_templates/test_weights_compression_backends.py]: extend to all criterion collectors

File: tests/openvino/native/quantization/test_weights_compression.py [hunk 4]
```diff
@@ -899,12 +899,55 @@ def test_compression_for_different_dtypes(activation_dtype, weight_dtype):
...
     ("compression_args", "multiplier_of_calls"),
-    (
-        (dict(mode=CompressWeightsMode.INT4_ASYM, ratio=1), 0),  # data-free, no reducers
-        (dict(mode=CompressWeightsMode.INT4_ASYM, ratio=0.5), 1),  # 1 reducer for mixed precision
-        (dict(mode=CompressWeightsMode.INT4_ASYM, ratio=1, awq=True), 2),  # mean & shape reducer for AWQ
-        (dict(mode=CompressWeightsMode.INT4_ASYM, ratio=0.5, awq=True), 3),  # 2 - for AWQ + 1 - for Mixed Precision
-    ),
+    [
+        ({"mode": CompressWeightsMode.INT4_ASYM, "ratio": 1}, 0),  # data-free, no reducers
+        ({"mode": CompressWeightsMode.INT4_ASYM, "ratio": 0.5}, 1),  # 1 reducer for mixed precision
```
Comment (ljaljushkin) [inline tests/openvino/native/quantization/test_weights_compression.py:904]: suppose the default one is not needed then

**Review by ljaljushkin — APPROVED** [review]

**Comment by ljaljushkin** [comment]
Could you run PTWC job, just in case?

**Comment by kshpv** [comment]
post_training_weight_compression 249 - in progress

File: tests/openvino/native/quantization/test_weights_compression.py (hunk 4 above)
Comment (kshpv) [reply tests/openvino/native/quantization/test_weights_compression.py]: done

**Review by nikita-savelyevv — APPROVED** [review]

**Pull request merged**